[docs][pigment] Add example and guide section (#41249)
mui/material-ui: packages/pigment-react/README.md

# Pigment CSS

> A Pigment CSS CSS-in-JS library that extracts the colocated styles to their own CSS files at build-time.

Reviewer: ```suggestion
Pigment CSS is a CSS-in-JS library that extracts the colocated styles to their own CSS files at build time.
```

  - [Color schemes](#color-schemes)
  - [Switching color schemes](#switching-color-schemes)
  - [TypeScript](#typescript)
- [How-to guides](#how-to-guides)
  - [Coming from Emotion or styled-components](#coming-from-emotion-or-styled-components)

## Getting started

Pigment CSS supports Next.js and Vite with support for more bundlers in future. You must install the corresponding plugin, as shown below.

### Next.js

#### Starter template

> Use the following commands to create a new Next.js project with Zero Runtime setup:

Reviewer: ```suggestion
Use the following commands to create a new Next.js project with Pigment CSS set up:
```

### Vite

#### Starter template

> Use the following commands to create a new Vite project with Zero Runtime setup:

Reviewer: ```suggestion
Use the following commands to create a new Vite project with Pigment CSS set up:
```

  }
}
```

## How-to guides

### Coming from Emotion or styled-components

> Emotion and styled-components are runtime CSS-in-JS libraries. What you write in your styles is what you get in the final bundle which means that the styles can be as dynamic as you want with the trade-offs of bundle size and performance overhead.

Reviewer: ```suggestion
Emotion and styled-components are runtime CSS-in-JS libraries. What you write in your styles is what you get in the final bundle, which means the styles can be as dynamic as you want with bundle size and performance overhead trade-offs.
```

> On the other hand, Pigment CSS extracts CSS at build time and replaces the JS code with hashed class names and some CSS variables. This means that it has to know all of the styles to be extracted ahead of time, so there are rules and limitations that you need to be aware of when using JavaScript callbacks or variables in Pigment CSS APIs.

Reviewer: ```suggestion
On the other hand, Pigment CSS, as a zero-runtime library, extracts CSS at build time and replaces the JS code with hashed class names and some CSS variables. This means that it has to know all of the styles to be extracted ahead of time, so there are rules and limitations that you need to be aware of when using JavaScript callbacks or variables in Pigment CSS's APIs.
```
